From e9f01660b650238d933b42f6c28d28d1c1872422 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 18:58:40 -0400
Subject: [fs] btrfs: reserve no transaction units in btrfs_feature_attr_store

Message-id: <1407179109-13714-34-git-send-email-sandeen@redhat.com>
Patchwork-id: 86985
O-Subject: [RHEL7.1 PATCH 033/422] btrfs: reserve no transaction units in btrfs_feature_attr_store
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: David Sterba <dsterba@suse.cz>

commit 43d87fa23154d135a2a1006bc6656ae73ae84190
Author: David Sterba <dsterba@suse.cz>
Date:   Mon Nov 18 14:24:20 2013 +0100

    btrfs: reserve no transaction units in btrfs_feature_attr_store

    Added in patch "btrfs: add ability to change features via sysfs",
    modifications to superblock don't need to reserve metadata blocks when
    starting a transaction.

    Signed-off-by: David Sterba <dsterba@suse.cz>
    Signed-off-by: Josef Bacik <jbacik@fb.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/sysfs.c b/fs/btrfs/sysfs.c
index 669fdf7..8fdc052 100644
--- a/fs/btrfs/sysfs.c
+++ b/fs/btrfs/sysfs.c
@@ -150,7 +150,7 @@ static ssize_t btrfs_feature_attr_store(struct kobject *kobj,
  btrfs_info(fs_info, "%s %s feature flag",
      val ? "Setting" : "Clearing", fa->kobj_attr.attr.name);
 
- trans = btrfs_start_transaction(fs_info->fs_root, 1);
+ trans = btrfs_start_transaction(fs_info->fs_root, 0);
  if (IS_ERR(trans))
   return PTR_ERR(trans);
 
-- 
1.7.1